lib/logmsg/type-hinting.c

```diff
@@ -166,22 +168,28 @@ _parse_fixed_point_timestamp_in_nsec(const gchar *value, gchar **endptr, gint64
         *nsec *= 10;
       return TRUE;
     }
-  return FALSE;
+  return TRUE;
 }
 
 gboolean
-type_cast_to_datetime_msec(const gchar *value, gint64 *out, GError **error)
+type_cast_to_datetime_unixtime(const gchar *value, UnixTime *ut, GError **error)
 {
   gchar *endptr;
   gint64 sec, nsec;
+  gint tzofs = -1;
 
   if (!_parse_fixed_point_timestamp_in_nsec(value, &endptr, &sec, &nsec))
     goto error;
 
-  if (*endptr != 0)
+  const guchar *tz_start = (guchar *) endptr;
+  gint tz_length = strlen(endptr);
+  if (*tz_start != 0 &&
+      !scan_iso_timezone(&tz_start, &tz_length, &tzofs))
```

Comment:
This adds an unnecessary `strlen()` call, when timezone info is not present. I think we should check `*tz_start != 0` first.

Reply:
reorganized, the strlen() is only done if *tz_start != 0

thanks.